ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,137 @@
+import json
+import logging
+from json import JSONDecodeError
+import requests
+
+from ocp_resources.route import Route
+from ocp_resources.secret import Secret
+from ocp_resources.service_account import ServiceAccount
+from ocp_resources.utils import TimeoutExpiredError, TimeoutSampler
+from ocp_utilities.infra import get_client
+
+TIMEOUT_10MIN = 10 * 60
+
+LOGGER = logging.getLogger(__name__)
+
+class Prometheus(object):
+    """
+    For accessing Prometheus cluster metrics
+
+    Prometheus HTTP API doc:
+    https://prometheus.io/docs/prometheus/latest/querying/api/
+
+    Argument for query method should be the entire string following the server address
+        e.g.
+        prometheus = Prometheus()
+        up = prometheus.query("/api/v1/query?query=up")
+    """
+
+    def __init__(
+        self,
+        namespace="openshift-monitoring",
+        resource_name="prometheus-k8s",
+        client=None,
+    ):
+        self.namespace = namespace
+        self.resource_name = resource_name
+        self.client = get_client()
+        self.api_v1 = "/api/v1"
+
+        # get route to prometheus HTTP api
+        self.api_url = self._get_route()
+
+        # get prometheus ServiceAccount token
+        self.headers = self._get_headers()
+
+    def _get_route(self):
+        # get route to prometheus HTTP api
+        LOGGER.info("Prometheus: Obtaining route")
+        route = Route(
+            namespace=self.namespace, name=self.resource_name, client=self.client
+        ).instance.spec.host
+
+        return f"https://{route}"
+
+    def _get_headers(self):
+        """Uses the Prometheus serviceaccount to get an access token for OAuth"""
+        LOGGER.info("Prometheus: Setting headers")
+
+        LOGGER.info("Prometheus headers: Obtaining OAuth token")
+
+        # get SA
```

Comment:
please avoid adding comments which describe the code; the code should be self-explanatory (applies to all comments)

Reply:
ack